editor/components/rich-text/index.js

```diff
@@ -773,20 +752,19 @@ export class RichText extends Component {
 			return accFormats;
 		}, {} );
 
-		let rect;
-		const selectedAnchor = find( parents, ( node ) => node.tagName === 'A' );
-		if ( selectedAnchor ) {
-			// If we selected a link, position the Link UI below the link
-			rect = selectedAnchor.getBoundingClientRect();
-		} else {
-			// Otherwise, position the Link UI below the cursor or text selection
-			rect = getRectangleFromRange( this.editor.selection.getRng() );
-		}
-		const focusPosition = this.getFocusPosition( rect );
-
-		this.setState( { formats, focusPosition, selectedNodeId: this.state.selectedNodeId + 1 } );
+		this.setState( { formats, selectedNodeId: this.state.selectedNodeId + 1 } );
 
 		if ( this.props.isViewportSmall ) {
+			let rect;
```

Comment:
Why is this needing to be calculated at every node change event? Do we need to scroll every time on that event? Cc @mcsf.

Reply:
It almost certainly doesn't need to be. There's definitely room for improvement here, but I considered it outside the scope of the current task.

Reply:
Sounds good, I like the smaller PRs :)

Reply:
I'm sure there's room for improvement. When we looked in #5769, there was no proper way to detect when the insertion of a character fills the whole width of a text container and forces the caret into a "new line" (a different height in the same container). `'nodechange'` is definitely too generic, but per the comment, _on `nodechange` we can work with the finalized TinyMCE instance and scroll to proper position._

Reply:
Perhaps the same kind of reasoning we're seeing in this PR (let's manage this ourselves with our vdom abstraction so as to batch positioning writes and reads) can be applied to the caret-tracking scrolling feature.